hydra-node/test/Hydra/BehaviorSpec.hs

```diff
@@ -48,294 +48,301 @@ import Test.Aeson.GenericSpecs (roundtripAndGoldenSpecs)
 import Test.Util (shouldBe, shouldNotBe, shouldReturn, shouldRunInSim, shouldSatisfy, traceInIOSim)
 
 spec :: Spec
-spec = describe "Behavior of one ore more hydra nodes" $ do
-  describe "Sanity tests of test suite" $ do
-    it "does not delay for real" $
-      failAfter 1 $ shouldRunInSim $ threadDelay 600
-
-    it "does detect when no responses are sent" $ do
-      let action = shouldRunInSim $ do
+spec = parallel $
+  describe "Behavior of one ore more hydra nodes" $ do
+    parallel $
+      describe "Sanity tests of test suite" $ do
+        it "does not delay for real" $
+          failAfter 1 $ shouldRunInSim $ threadDelay 600
+
+        it "does detect when no responses are sent" $ do
+          let action = shouldRunInSim $ do
+                chain <- simulatedChainAndNetwork
+                withHydraNode 1 [] NoSnapshots chain $ \n ->
+                  waitForNext n >> failure "unexpected output"
+          action `shouldThrow` \case
+            FailureDeadlock _ -> True
+            _ -> False
+
+    parallel $
```

Comment:
Question: Do we need `parallel` at multiple levels? Isn't a single `parallel` at top-level enough?

Reply:
The docs says `parallel marks all spec items of the given spec to be safe for parallel evaluation.`. Now, it's a bit unclear what an _items_ of a spec is, and if it's only the direct descendant or also the nested children.

Reply:
According to source code it's applied recursively: https://github.com/hspec/hspec/blob/b48eee41032450123044de82c4c02213813563b1/hspec-core/test/Test/Hspec/Core/SpecSpec.hs#L111